### DISCUSSION TITLE
[Feat] Allow users to abort running commands

### DISCUSSION THREAD

**Discussion by albertodlc** [sourcegit-scm/sourcegit#753] [Ideas]
Allow users to cancel running commands to avoid waiting until the error message is displayed.

**Example**
I tried to push a commit and I didn't have connectivity with the remote, so I had to wait until Sourcegit return an error
![image](https://github.com/user-attachments/assets/a1b1ef8f-7b29-4de0-8501-8c64da60687a)
![image](https://github.com/user-attachments/assets/3419242e-cefd-4916-819a-e10dc417829d)

**Comment by love-linger** [comment]
It's not safe to cancel a running git command directly. You can use `git config --global http.timeout <SECONDS>` to reduce the waiting time of network connection.

**Comment by aikawayataro** [comment]
I think it is safe to send `SIGTERM` to gracefully stop git as it installs a handler for this. The problem is there's no dotnet API for such thing https://github.com/dotnet/runtime/issues/109432

**Comment by albertodlc** [comment]
[GitExtension](https://github.com/gitextensions/gitextensions) allow this behavior. And it is based on dotnet (I think) so maybe we can replicate their approach.

**Comment by aikawayataro** [comment]
> [GitExtension](https://github.com/gitextensions/gitextensions) allow this behavior. And it is based on dotnet (I think) so maybe we can replicate their approach.

Their approach is Windows-only

It is possible to use p/invoke and call whatever we need for the corresponding OS but that's cumbersome

**Comment by dougcunha** [comment]
I also intended to propose this feature.  I've experienced significant delays cloning GitHub repositories in SourgeGit, with no option to cancel except by forcefully terminating the SourgeGit processes. Although the cloning issue can be problematic in itself, a dedicated cancel button would be a valuable addition.

**Comment by p-fischer** [comment]
I'm also in favor of this proposal. 
I regularly run into this problem when triggering fetch/pull/push on a repository, which requires a VPN being connected. If I forgot that and establish the VPN connection shortly after triggering e.g. fetch, then I still have to wait for quite a while (like a connectivity timeout).
My workaround is to quit SourceGit and restart, which doesn't seem ideal.

**Reply by arkanoider** [reply]
I think i have a similar issue also if on a push of a commit i flag make sure `submodules have been pushed` , but they are not. It signals abort, but never exit.
